zulipterminal/ui_tools/views.py

```diff
@@ -436,7 +447,7 @@ def __init__(
             header=self.header_list,
         )
         self.search_lock = threading.Lock()
-        self.empty_search = False
```

Comment:
I think it's not on this PR to solve this, but it's easy to notice here: It's unfortunate that there's a lot of duplication between all the different views. A similar logic change repeated three times.

I'm currently not sure what could be a good way to refactor this. Maybe a class that knows the view state, handles the keypresses, and forwards back to the view what should happen based on some interface.